plugins/secureboot-paper/src/main/java/net/civmc/secureboot/paper/SecureBootPlugin.java

```diff
@@ -0,0 +1,132 @@
+    private boolean acceptLogins = false;
+
+    @Override
+    public void onLoad() {
+        saveDefaultConfig();
+        // ensure required plugins have a status
+        getConfig().getStringList("required_plugins").forEach((plugin) -> this.pluginState.put(plugin, PluginStatus.AWAITING_STATUS));
+        getSLF4JLogger().debug("Awaiting status for: {}", String.join(", ", this.pluginState.keySet()));
+
+        // set kick message from config
+        String rawKickMessage = getConfig().getString("kick_message", "Server is in maintenance mode.");
+        kickMessage = mm.deserialize(rawKickMessage);
+    }
+
+    @Override
+    public void onEnable() {
+        Bukkit.getPluginManager().registerEvents(this, this);
+
+        // add all plugins loaded by the server
+        for (final Plugin plugin : Bukkit.getPluginManager().getPlugins()) {
+            this.pluginState.replace(
+                plugin.getPluginMeta().getName(),
+                PluginStatus.AWAITING_STATUS,
+                // Just in case this plugin was enabled prior to SecureBoot
+                plugin.isEnabled() ? PluginStatus.ENABLED : PluginStatus.LOADED
+            );
+        }
+    }
+
+    /**
+     * This is emitted <i>after</i> the plugin's {@link org.bukkit.plugin.Plugin#onEnable()} method has been called,
+     * but not if it threw. This does, however, require plugins to fail exceptionally to be detected as having not
+     * started correctly.
+     */
+    @EventHandler(priority = EventPriority.MONITOR)
+    private void detectPluginEnabled(
+        final @NotNull PluginEnableEvent event
+    ) {
+        // mark enabled plugin as enabled
+        this.pluginState.replace(event.getPlugin().getPluginMeta().getName(), PluginStatus.LOADED, PluginStatus.ENABLED);
+    }
+
+    /**
+     * This is emitted <i>before</i> the plugin's {@link org.bukkit.plugin.Plugin#onDisable()} method is called, so no
+     * failure on the plugin's part to properly disable itself should interfere with this.
+     */
+    @EventHandler(priority = EventPriority.MONITOR)
+    private void detectPluginDisabled(
+        final @NotNull PluginDisableEvent event
+    ) {
+        // mark disabled plugin as loaded
+        this.pluginState.replace(event.getPlugin().getPluginMeta().getName(), PluginStatus.ENABLED, PluginStatus.LOADED);
+    }
+
+    // need event to run as late as possible to catch any lingering issues
+    @EventHandler(priority = EventPriority.MONITOR)
+    private void onServerLoad(
+        final @NotNull ServerLoadEvent event
+    ) {
+        var foundError = false;
+        for (final Map.Entry<String, PluginStatus> entry : this.pluginState.entrySet()) {
+            var pluginName = entry.getKey();
+            foundError |= switch (entry.getValue()) {
+                case ENABLED -> {
+                    getSLF4JLogger().debug("{} loaded successfully!", pluginName);
```

Comment:
remove .debug it won't do anything

Reply:
Can't you set the log level to debug? I feel like this is something worth logging, even if only in debug